src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        BulldozerProducer bulldozerProducer = new BulldozerProducer();
+        TruckProducer truckProducer = new TruckProducer();
+        ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+        List<? extends Machine> machines = new ArrayList<>();
+        if (type == Bulldozer.class) {
+            machines = bulldozerProducer.get();
+        } else if (type == Excavator.class) {
+            machines = excavatorProducer.get();
+        } else if (type == Truck.class) {
+            machines = truckProducer.get();
+        }
+        return new ArrayList<>(machines);
+    }
```

Comment:
maybe you can create new object only if if-statment true. I mean 
if (type == Bulldozer.class) {
 BulldozerProducer bulldozerProducer = new BulldozerProducer();
            machines = bulldozerProducer.get(); 
} else ...